lib/collection/src/collection/point_ops.rs

```diff
@@ -419,6 +451,14 @@ impl Collection {
                         if shard_key.is_none() {
                             return Ok(records);
                         }
+
+                        // If resharding, exclude migrated points from old shards
+                        if let Some(filter) = resharding_filter.as_ref() {
+                            if do_filter {
+                                records.retain(|record| !filter.check(record.id));
+                            }
+                        }
```

Comment:
Also, I'm not 100% sure, but I think we might have to filter points *before* the `shard_key.is_none()` early return.